pkg/apiserver/statement/queries.go

```diff
@@ -137,15 +137,24 @@ func (s *Service) queryPlans(
 		return nil, err
 	}
 
-	err = db.
+	query := db.
 		Select(selectStmt).
 		Table(statementsTable).
 		Where("summary_begin_time >= FROM_UNIXTIME(?) AND summary_end_time <= FROM_UNIXTIME(?)", beginTime, endTime).
-		Where("schema_name = ?", schemaName).
-		Where("digest = ?", digest).
-		Group("plan_digest").
-		Find(&result).
-		Error
+		Group("plan_digest")
+
+	if schemaName != "" {
+		query.Where("schema_name = ?", schemaName)
+	}
+	if digest == "" {
```

Comment:
So frontend will pass `digest=""` to the backend, even if the digest from the backend is null?

Reply:
Go: 🤧

Reply:
Yeah, there are two things we need to improve:
- Define a new structure. Embed `sql.NullString` to scan NULL value from db and customize the MashallJSON/UnmashallJSON interface to serde null value
- The get request is not easy to carry null, we can use undefined to not pass this parameter

But I don't think that's going to help here. I reverted the code, LOL.

Reply:
Because `is null` and `= ''` are always two patterns.